arrow-arith/src/boolean.rs

```diff
@@ -332,7 +339,14 @@ pub fn is_null(input: &dyn Array) -> Result<BooleanArray, ArrowError> {
 /// ```
 pub fn is_not_null(input: &dyn Array) -> Result<BooleanArray, ArrowError> {
     let values = match input.nulls() {
-        None => BooleanBuffer::new_set(input.len()),
+        None => {
+            // NullArray has no nulls buffer yet all values are null
+            if input.data_type().equals_datatype(&DataType::Null) {
```

Comment:
```suggestion
            if input.data_type() == &DataType::Null {
```
Could also possibly be lifted into the match expression

Reply:
Sure, done (both)